magicmaze.js

```diff
@@ -657,21 +657,55 @@ function drawMagePreviews (obj) {
   }
 }
 
-function placeCharacter (obj, info) {
+
+
+function placeCharacter (obj, info, warp = false) {
   const x = parseInt(info.position_x)
   const y = parseInt(info.position_y)
   const tokenId = parseInt(info.token_id)
+  const [oldx, oldy] = obj.characterLocs.get(tokenId) ?? [x, y]
   obj.characterLocs.set(tokenId, [x, y])
   const key = getKey(x, y)
   const top = obj.tops.get(key)
   const left = obj.lefts.get(key)
   const adjust = (CELL_SIZE - MEEPLE_SIZE) / 2
+  var halftime = 100
```

Comment:
we never use halftime in a context other than some multiplier * halftime? should this just be the fulltime?

(also you can make it `const animTime = warp ? 600 : 200` or something)